### PR TITLE
doc/architecture: edit "HA Authentication"

### PR DESCRIPTION
**Pull request by zdover23**
Edit "High Availability Authentication" in doc/architecture.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -318,9 +318,10 @@ and uses it to sign requests to OSDs and to metadata servers in the cluster.
 
 
 The ``cephx`` protocol authenticates ongoing communications between the client
-machine and the Ceph servers. Each message sent between a client and server,
-subsequent to the initial authentication, is signed using a ticket that the
-monitors, OSDs and metadata servers can verify with their shared secret.
+machine and the Ceph servers. After initial authentication, each message sent
```
Comment (zdover23) [inline doc/architecture.rst:321]: @anthonyeleven: What do you think of this word, "servers"? It seems to me that we often say "hosts" or "nodes" in our docs, and suddenly here we are busting out a third synonym just to confuse our readers. What do you think?

Comment (anthonyeleven) [inline doc/architecture.rst:321]: s/machine//
s/and the/and/

the second is marginal

Comment (zdover23) [reply doc/architecture.rst:321]: Accepted.

Comment (zdover23) [reply doc/architecture.rst:321]: Well, I'm leaving the "the" in the second part for the sake of completely parallel construction.

Comment (anthonyeleven) [reply doc/architecture.rst]: Why stop confusing our readers now? ;). 

I suggest 
>between clients and Ceph daemons.

File: doc/architecture.rst [hunk 2]
```diff
@@ -356,11 +357,10 @@ monitors, OSDs and metadata servers can verify with their shared secret.
                 |<-------------------------------------------|
                                receive response
 
-The protection offered by this authentication is between the Ceph client and the
-Ceph server hosts. The authentication is not extended beyond the Ceph client. If
-the user accesses the Ceph client from a remote host, Ceph authentication is not
-applied to the connection between the user's host and the client host.
-
+This authentication protects only the connection between Ceph clients and Ceph
+server hosts. The authentication is not extended beyond the Ceph client. If a
```
Comment (anthonyeleven) [inline doc/architecture.rst:361]: and Ceph daemons.

File: doc/architecture.rst [hunk 3]
```diff
@@ -356,11 +357,10 @@ monitors, OSDs and metadata servers can verify with their shared secret.
...
+user accesses the Ceph client from a remote host, Ceph authentication will not
```
Comment (anthonyeleven) [inline doc/architecture.rst:362]: , cephx authentication

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst]: This is exactly what I was hoping that you'd suggest, in exactly the words I was hoping that you'd suggest it.

Comment (anthonyeleven) [reply doc/architecture.rst]: :p

Comment (anthonyeleven) [reply doc/architecture.rst]: :p

Comment (anthonyeleven) [reply doc/architecture.rst]: One might then wonder why you didn't just go with that in the first place ;)

Comment (zdover23) [reply doc/architecture.rst]: Subsequent developments related to the incorporation of the word "daemon" where previously "server" was have obviated the need for this definite article. To the degree that there was any agon in this matter, you can consider this a victory.

Comment (zdover23) [reply doc/architecture.rst]: I wanted to keep your expertise free of the taint of any suggestion from me, so that it was pure when it was tapped.

Comment (anthonyeleven) [reply doc/architecture.rst]: I HAVE TEH_POWER

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53632 - Reef backport
https://github.com/ceph/ceph/pull/53633 - Quincy backport

**Pull request merged**